frontend/src/routers/AppRouter/AppRouter.tsx

```diff
@@ -115,13 +124,30 @@ export const AppRouter = () => {
 	const isNewWorkspacePage = !!newProjectMatch
 	const isJoinWorkspacePage = !!joinWorkspaceMatch
 	const [inviteCode, setInviteCode] = useLocalStorage('highlightInviteCode')
+	const [lastVisitedRoute, setLastVisitedRoute] =
+		useLocalStorage('lastVisitedRoute')
 	const { projectId } = useNumericProjectId(previousLocation)
 	const [nextParam] = useQueryParam('next', StringParam)
 	const [configurationIdParam] = useQueryParam('configurationId', StringParam)
 	const isVercelIntegrationFlow = !!nextParam || !!configurationIdParam
 	const navigate = useNavigate()
 	const isValidProjectId = Number.isInteger(Number(projectId))
 
+	useEffect(() => {
+		if (isValidLastVisitedRoute(location.pathname)) {
+			setLastVisitedRoute(location.pathname)
```

Comment:
this may end up saving filters into the last visited url that will be reapplied, creating a confusing experience. for instance https://app.highlight.io/1/logs/MjAyNS0wMS0yMVQxNjoxMTozNVosNGY2M2NmNTYtZWJiMi00ZTNlLTgyNzQtNWVkODlkMDIwYjU1 will be saved with the specific log line from the `view in context` button because the log is part of the path rather than query string (i noticed you are already correctly dropping the QS to avoid saving other filter params like https://preview.highlight.io/1/sessions?end_date=2025-01-08T05%3A59%3A59.999Z&start_date=2025-01-06T06%3A00%3A00.000Z). 

since this is specific to the products, perhaps we truncate the pathname as well to just save the `/sessions`, `/errors`, etc. value without a sub-path?